✨ 상세페이지 axios 인스턴스 생성 및 mock API 연결

yongarframe opened the pull request:
## ✅ PR 요약

- 관련 이슈 번호 : #104 
- 작업요약 : axios인스턴스 생성 및 상세페이지 mock API 연결

<!-- ex: feat: 로그인 페이지 구현 -->

## 📋 상세 내용

<!-- 어떤 작업을 했는지 간단히 설명 -->

- axios 인스턴스 생성
- 커뮤니티 상세페이지 mockAPI 연결
- 댓글 부분 mockAPI 연결
- API 데이터에 맞게 type 수정

## 📸 스크린샷 (선택)

<!-- UI 변경 시 첨부 -->

## 📝 기타 참고사항

## 🧪 PR Check

<!-- 직접 테스트한 내용, 어떻게 동작을 확인했는지 작성 -->

- [x] 정상 동작 확인
- [x] UI 렌더링 확인
- [x] 기능 동작 확인
- [x] lint, type-check, build 오류 확인

yongarframe committed:
✨ feat : 커뮤니티 상세페이지 데이터 fetch 로직 구현 및 연동 완료

yongarframe committed:
✨ feat : 작성 날짜를 ~~시간/일/달/년 전 작성 으로 변경

yongarframe committed:
✨ feat : 댓글 부분 mock API 연결

chen4023 commented on src/api/mainApi.ts:
```diff
@@ -0,0 +1,12 @@
+import axios from 'axios'
+
+// 인스턴스 생성
+const axiosMainApiInstance = axios.create({
```
여러가지 인스턴스를 생성하실 목적이 아니라면 변수명을 api와 같이 간략하게 지정하셔도 좋을 것 같아요!

chen4023 commented on src/api/mainApi.ts:
```diff
@@ -0,0 +1,12 @@
+  },
+})
+
+export default axiosMainApiInstance
```
엑세스 토큰을 요청에 포함하기 위한 인터셉터도 만들어두신 인스턴스에 함께 적용하는 방법도 고려해보시면 좋을 것 같네요!

chen4023 commented on src/components/commnunityDetail/Comment.tsx:
```diff
@@ -1,19 +1,20 @@
 import CommonModal from '@components/common/Modal'
 import { Button } from '@components/common'
 import { commentsMockData } from './mockData'
+import photo from '../../assets/profile_default.png'
```
절대경로가 적용되지 않은 것 같아요!

chen4023 commented on src/components/commnunityDetail/Comment.tsx:
```diff
@@ -32,10 +33,15 @@ export default function Comment({
   return (
     <div className="flex gap-[17px] w-full relative">
-      <img src={imgUrl} className="w-[48px] h-[48px] rounded-[50%] " />
+      <img
```
`img` 태그를 사용하는 경우에는 웹접근성을 위해 `alt`속성을 지정해주시면 좋습니다!
[참고용](https://velog.io/@eunsoo0504/img%ED%83%9C%EA%B7%B8-alt-%EC%86%8D%EC%84%B1-%EC%8D%A8%EC%95%BC%ED%95%A0%EA%B9%8C)

chen4023 reviewed (commented):
고생하셨습니다 용훈님!
인스턴스를 생성하신 경우, 인터셉터도 한 번에 적용해주시면 좋을 것 같네요!👍🏻
추가적으로 변수명이 변경되는 경우가 많아보이는데 특별한 이유가 있을까요?

yongarframe commented:
> 고생하셨습니다 용훈님! 인스턴스를 생성하신 경우, 인터셉터도 한 번에 적용해주시면 좋을 것 같네요!👍🏻 추가적으로 변수명이 변경되는 경우가 많아보이는데 특별한 이유가 있을까요?

1) CommentsInfiniteScroll.tsx --> comment.tsx 에서 내려주는 prop 이 commentData 에서 comment 로 변경되었는데 좀더 직관적인 변수명인거 같아서 적용 하였습니다.

2) comment 의 타입 commentData 는 s를 붙였는데 잘못 붙인거 같아 수정 하였습니다.

3) postData 는 글삭제/수정을 담당자가 상세페이지와 연결하면서 변수명을 작성해서 detail 페이지에 맞게 변수명 적용 하였습니다.

yongarframe committed:
♻️ refactor : 변수명 수정 - axios 인스턴스 변수 명 수정 및 comment 의 타입 변수명 수정

axios BaseUrl 적용되는 변수명 수정 : axiosMainApiInstance -> api
comment 타입의 변수명 수정 : CommentsData에서 s 삭제 후 CommentData로 적용
